### PR TITLE
[3.1][Kernel] Downgrade SLF4J version to 1.x to avoid compatibility issues with connectors

### PR DESCRIPTION
**Pull request by allisonport-db**
#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [X] Kernel
- [ ] Other (fill in here)

## Description

Downgrades SLF4J version to 1.x. Version 2.x includes additional APIs that are unsupported in lower versions. This avoids using those APIs.

## How was this patch tested?

 Ran some unit tests and checked the output logs locally.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -246,13 +246,13 @@ lazy val kernelApi = (project in file("kernel/kernel-api"))
     Test / javaOptions ++= Seq("-ea"),
     libraryDependencies ++= Seq(
       "org.roaringbitmap" % "RoaringBitmap" % "0.9.25",
-      "org.slf4j" % "slf4j-api" % "2.0.9",
+      "org.slf4j" % "slf4j-api" % "1.7.36",
```
Comment (allisonport-db) [inline build.sbt:249]: Chose based on "older stable version" here https://www.slf4j.org/download.html

File: kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java [hunk 2]
```diff
@@ -473,24 +471,22 @@ protected Optional<LogSegment> getLogSegmentForVersion(
...
+        logDebug(() ->
             String.format(
                 "deltasAfterCheckpoint: %s",
                 Arrays.toString(deltasAfterCheckpoint.stream().map(
-                    x -> new Path(x.getPath()).getName()).toArray())
-            )
-        ).log();
+                    x -> new Path(x.getPath()).getName()).toArray())));
 
-        // todo again naming confusing (specify after checkpoint?)
```
Comment (allisonport-db) [inline kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java:484]: I think I accidentally merged this TODO in a previous PR...

**Review by nicklan — APPROVED** [review]
lgtm!

**Review by vkorukanti — APPROVED** [review]
lgtm

File: kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java [hunk 3]
```diff
@@ -580,4 +576,11 @@ private Optional<LogSegment> getLogSegmentWithMaxExclusiveCheckpointVersion(
         // TODO
         return Optional.empty();
     }
+
+    // TODO logger interface to support this across kernel-api module
+    private void logDebug(Supplier<String> message) {
```
Comment (scottsand-db) [inline kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java:581]: nice!

**Review by scottsand-db — APPROVED** [review]

File: kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java (hunk 3 above)
Comment (tdas) [reply kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java:581]: +1 lets make this interface. its going to be internal interface right? so its okay if this is dont after 3.1.

**Review by tdas — APPROVED** [review]

**Pull request merged**